arcgis-ios-sdk-samples/Route & Navigation/Route around barriers/RouteAroundBarriersViewController.swift

```diff
@@ -121,36 +121,36 @@ class RouteAroundBarriersViewController: UIViewController, AGSGeoViewTouchDelega
         self.routeParameters.clearPolygonBarriers()
         self.routeParameters.setPolygonBarriers(barriers)
         
-        self.routeTask.solveRouteWithParameters(self.routeParameters) { [weak self] (routeResult:AGSRouteResult?, error:NSError?) -> Void in
+        self.routeTask.solveRoute(with: self.routeParameters) { [weak self] (routeResult:AGSRouteResult?, error:Error?) -> Void in
             if let error = error {
-                SVProgressHUD.showErrorWithStatus("\(error.localizedDescription) \(error.localizedFailureReason ?? "")")
+                SVProgressHUD.showError(withStatus: "\(error.localizedDescription) \((error as NSError).localizedFailureReason ?? "")")
             }
             else {
                 SVProgressHUD.dismiss()
                 let route = routeResult!.routes[0]
                 let routeGraphic = AGSGraphic(geometry: route.routeGeometry, symbol: self!.routeSymbol(), attributes: nil)
-                self?.routeGraphicsOverlay.graphics.addObject(routeGraphic)
+                self?.routeGraphicsOverlay.graphics.add(routeGraphic)
                 self?.generatedRoute = route
             }
         }
     }
     
     func routeSymbol() -> AGSSimpleLineSymbol {
-        let symbol = AGSSimpleLineSymbol(style: .Solid, color: UIColor.yellowColor(), width: 5)
+        let symbol = AGSSimpleLineSymbol(style: .solid, color: UIColor.yellow, width: 5)
         return symbol
     }
     
     func directionSymbol() -> AGSSimpleLineSymbol {
-        let symbol = AGSSimpleLineSymbol(style: .DashDot, color: UIColor.orangeColor(), width: 5)
+        let symbol = AGSSimpleLineSymbol(style: .dashDot, color: UIColor.orange, width: 5)
         return symbol
     }
     
-    private func symbolForStopGraphic(index: Int) -> AGSSymbol {
+    fileprivate func symbolForStopGraphic(_ index: Int) -> AGSSymbol {
```

Comment:
similar to other comments above about having argument name to clarify call site usage